assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -451,9 +452,18 @@ public void handleAccess(HttpServletRequest req, HttpServletResponse res, Refere
                     // fail the request, user not logged in yet.
                 } else {
                     if (REF_TYPE_SUBMISSIONS.equals(ref.getSubType())) {
+                        String assignmentName = "bulk_download";
+                        try {
+                            assignmentName = getAssignment(ref.getReference()).getTitle();
```

Comment:
have you checked if you can get the title from the ref?

Reply:
No I have not.